Add custom pull step examples with variables (#10073)
PrefectHQ/prefect: docs/concepts/deployments-ux.md

        stream_output: False
```

> Below is an example that retrieves an access token from a 3rd party keyvault and uses it for a private clone step:

Reviewer: ```suggestion
Below is an example that retrieves an access token from a 3rd party Key Vault and uses it in a private clone step:
```

```yaml
pull:
- prefect.deployments.steps.run_shell_script:
    id: get-access-token
    script: az keyvault secret show --name <secret name> --vault-name <secret vault> --query "value" --output tsv
    stream_output: false
- prefect.deployments.steps.git_clone:
    repository: https://bitbucket.org/samples/deployments.git
    branch: master
    access_token: "{{ get-access-token.stdout }}"
```

> An additional way to run custom steps that are not defined is provided below. `retrieve_secrets` is a custom python module that has been packaged into the default working directory of a docker image (which is /opt/prefect by default). `main` is the function entry point, which returns an access token (e.g. `return {"access_token": access_token}`) like the preceding example, but utilizing the Azure Python SDK for retrieval. 

Reviewer: ```suggestion
You can also run custom steps by packaging them. In the example below, `retrieve_secrets` is a custom python module that has been packaged into the default working directory of a docker image (which is /opt/prefect by default). `main` is the function entry point, which returns an access token (e.g. `return {"access_token": access_token}`) like the preceding example, but utilizing the Azure Python SDK for retrieval.
```
